content/shmem_malloc_hints.tex

```diff
@@ -0,0 +1,116 @@
+
+\apisummary{
+    Collective memory allocation routine with support for providing hints.
+}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+void *@\FuncDecl{shmem\_malloc\_with\_hints}@(size_t size, long hints);
+\end{Csynopsis}
+
+\begin{apiarguments}
+    \apiargument{IN}{size}{The size, in bytes, of a block to be
+        allocated from the symmetric heap. This argument is of type \CTYPE{size\_t}}
+    \apiargument{IN}{hints}{A bit array of hints provided by the user to the implementation}
+\end{apiarguments}
+
+
+\apidescription{
+
+    The \FUNC{shmem\_malloc\_with\_hint} like \FUNC{shmem\_malloc} routine returns a pointer to a block of at least
+    \VAR{size} bytes, which shall be suitably aligned so that it may be
+    assigned to a pointer to any type of object.  This space is allocated from
+    the symmetric heap (similar to \FUNC{shmem\_malloc}).  When \VAR{size} is zero, 
+    the \FUNC{shmem\_malloc\_with\_hint} routine performs no action and returns a null pointer. 
+    
+    In addition to the \VAR{size} argument, the \VAR{hint} argument is provided by the user. 
+    The \VAR{hint} could describe the trait of memory storage where memory is allocated or 
+    it could describe the expected manner in which the \openshmem program may use the allocated memory.
+    The valid storage trait hints are described in Table~\ref{traithints} and the valid usage hints 
+    are described in Table~\ref{usagehints}. Multiple hints are expressed as \CONST{OR} of \VAR{hints}.
```

Comment:
@manjugv 
question regarding ```Multiple hints are expressed as OR of hints.```: how oshmem should react on case if supported less flags than requested? or supported all honts, but not at same buffer (and I believe this case - supported all but not at same buffer - is the most actual)?

it seems ```OR``` combination of ```traithints``` hints should be denied

Reply:
@hoopoepg That is correct. In the last sentence of this paragraph I say that. Some combination of hints does not make sense and some combination of hints might not be supported by a particular implementation. In either case, it can behave like shmem_malloc.